### PR TITLE
Scripting System Overhaul

### PR DESCRIPTION
**Pull request by shashagit**
-Array of Scripts in Script component
-Keyboard and controller events accessible in Scripts

### CODE REVIEW

File: HollowEngine/BulletHell/src/Systems/MovementSystem.cpp [hunk 1]
```diff
@@ -91,13 +91,13 @@ namespace BulletHell
...
+			lua["isMoveForward"] = Hollow::InputManager::Instance().GetAxisValue(SDL_CONTROLLER_AXIS_LEFTY) < -16000;
 
-			lua["isMoveBackward"] = (Hollow::InputManager::Instance().IsKeyPressed(SDL_SCANCODE_DOWN) || Hollow::InputManager::Instance().GetAxisValue(SDL_CONTROLLER_AXIS_LEFTY) > 16000);
+			lua["isMoveBackward"] = Hollow::InputManager::Instance().GetAxisValue(SDL_CONTROLLER_AXIS_LEFTY) > 16000;
 
-			lua["isStrafeLeft"] = (Hollow::InputManager::Instance().IsKeyPressed(SDL_SCANCODE_LEFT) || Hollow::InputManager::Instance().GetAxisValue(SDL_CONTROLLER_AXIS_LEFTX) < -16000);
+			lua["isStrafeLeft"] = Hollow::InputManager::Instance().GetAxisValue(SDL_CONTROLLER_AXIS_LEFTX) < -16000;
 
-			lua["isStrafeRight"] = (Hollow::InputManager::Instance().IsKeyPressed(SDL_SCANCODE_RIGHT) || Hollow::InputManager::Instance().GetAxisValue(SDL_CONTROLLER_AXIS_LEFTX) > 16000);
+			lua["isStrafeRight"] = Hollow::InputManager::Instance().GetAxisValue(SDL_CONTROLLER_AXIS_LEFTX) > 16000;
```
Comment (sidu7) [inline HollowEngine/BulletHell/src/Systems/MovementSystem.cpp]: shouldn't these use strings like RX LX?

File: HollowEngine/HollowEngine/src/Hollow/Components/Script.cpp [hunk 2]
```diff
@@ -18,19 +17,35 @@ namespace Hollow {
 
 	void Script::DebugDisplay()
 	{
-		ImGuiHelper::InputText("Script File", mScriptPath);
+		ImGui::InputInt("Count", &mScriptCount);
+		for (int i = 0; i < mScriptCount; ++i)
+		{
```
Comment (sidu7) [inline HollowEngine/HollowEngine/src/Hollow/Components/Script.cpp]: Need to make some changes to make it work with AddComponent in Editor

File: HollowEngine/BulletHell/src/Systems/MovementSystem.cpp (hunk 1 above)
Comment (drasske) [reply HollowEngine/BulletHell/src/Systems/MovementSystem.cpp]: This whole chunk of code will be moved to the player movement script

File: HollowEngine/HollowEngine/src/Hollow/Managers/ScriptingManager.cpp [hunk 3]
```diff
@@ -40,6 +70,7 @@ namespace Hollow
 			"__sub", [](const glm::vec2& l, const glm::vec2& r) { return glm::vec2(l.x - r.x, l.y - r.y); }
 			);
 
+		// COMPONENTS
 		lua.new_usertype<Body>("RigidBody",
```
Comment (drasske) [inline HollowEngine/HollowEngine/src/Hollow/Managers/ScriptingManager.cpp]: Whenever we want a new function in a script should we just add the lua binding in this file?

Comment (drasske) [reply HollowEngine/HollowEngine/src/Hollow/Managers/ScriptingManager.cpp]: Yes

**Pull request merged**